pkg/apis/experimental/types.go

```diff
@@ -464,9 +464,9 @@ type JobCondition struct {
 	Message string `json:"message,omitempty"`
 }
 
-// An Ingress is a way to give services externally-reachable urls. Each Ingress is a
-// collection of rules that allow inbound connections to reach the endpoints defined by
-// a backend.
+// An Ingress is a way to give services externally-reachable urls. Each Ingress
+// is a collection of rules that allow inbound connections to reach the endpoints
+// defined by a backend.
 type Ingress struct {
```

Comment:
Seems pretty obvious by the comment that it's trying to allow external-to-internal access. Tho if you have subdivision of internal networks along namespaces you might also need it for internal pods to talk to services in another namespaces. Internal-to-internal technically isn't really Egress tho, even if you have to go out through some non-k8s networking or load balancer.

I think the primary use case was for the networking abstraction layer that k8s sits on to create load balancers, network routes, proxies, and/or port forwarding for external-to-internal traffic.